src/transaction/confirm.rs

```diff
@@ -199,12 +199,12 @@ impl<T: Transport> ConfirmationContext<'_, T> {
     /// Waits for a certain number of blocks to be mined using a block filter.
     async fn wait_for_blocks_with_filter(&self, block_count: usize) -> Result<(), ExecutionError> {
         let block_filter = self.web3.eth_filter().create_blocks_filter().await?;
-        let mut stream = block_filter.stream(self.params.poll_interval);
-        for _ in 0..block_count {
-            stream
-                .next()
-                .await
-                .ok_or(ExecutionError::StreamEndedUnexpectedly)??;
+        let stream = block_filter.stream(self.params.poll_interval);
+        let results = stream.take(block_count).collect::<Vec<_>>().await;
+        let blocks = results.into_iter().collect::<Result<Vec<_>, _>>()?;
+
+        if blocks.len() < block_count {
+            return Err(ExecutionError::StreamEndedUnexpectedly);
         }
```

Comment:
This should be
```suggestion
        let stream = block_filter.stream(self.params.poll_interval);
        futures::pin_mut!(stream);
        for _ in 0..block_count {
            stream
                .next()
                .await
                .ok_or(ExecutionError::StreamEndedUnexpectedly)??;
        }
```
The only new line compared to master is `futures::pin_mut!(stream);`.
The reason we need this is that internally web3 no longer guarantees that the stream is Unpin.

Reply:
You are right. I was trying to remove this requirement reworking how to work was being done. But I just reverted the changes and added the pin mut like you suggested! (: PR updated